### PR TITLE
add Ubuntu support for kube-up.sh

### PR DESCRIPTION
**Pull request by sonyafenge**
**What type of PR is this?**

> /kind feature

**What this PR does / why we need it**:
Add python3 installation to support ubuntu for kube-up.sh. 

**Existing issue for this PR**:
Cluster can be started with PR, but if cni plugin using flannel, several pods failed to start: metrics-server, kubernetes-dashboard, event-exporter, corednns. There will be a follow-up PR to fix this issue.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by sonyafenge** [comment]
investigation for flannel plugin is in progress. This PR can be used for Mizar integration.

File: cluster/addons/dns/coredns/coredns.yaml.base [hunk 1]
```diff
@@ -129,6 +129,8 @@ spec:
         - name: config-volume
           mountPath: /etc/coredns
           readOnly: true
+        - name: tmp
```
Comment (h-w-chen) [inline cluster/addons/dns/coredns/coredns.yaml.base:132]: is there a reason we need to explicitly define /tmp here?

File: cluster/gce/gci/configure.sh [hunk 2]
```diff
@@ -56,13 +56,15 @@ EOF
 }
 
 function validate-python {
-  local ver=$(python -c"import sys; print(sys.version_info.major)")
-  echo "python version: $ver"
-  if [[ $ver -ne 2 && $ver -ne 3 ]]; then
+  local ver=$(python3 -c"import sys; print(sys.version_info.major)")
```
Comment (h-w-chen) [inline cluster/gce/gci/configure.sh:59]: 1. isn't python to be installed for ubuntu 20.04+ version 3?
2. besides this imperative installation, is it possible to leverage cloud-init way?

File: cluster/gce/gci/scaleoutserver-helper.sh [hunk 3]
```diff
@@ -85,7 +85,7 @@ function create-proxy-instance-internal() {
       --zone "${ZONE}" \
       --machine-type "${MASTER_SIZE}" \
       --image-project="${PROXY_IMAGE_PROJECT:-ubuntu-os-cloud}" \
-      --image "${PROXY_IMAGE:-ubuntu-1804-bionic-v20201014}" \
+      --image "${PROXY_IMAGE:-ubuntu-2004-focal-v20211202}" \
```
Comment (h-w-chen) [inline cluster/gce/gci/scaleoutserver-helper.sh:88]: if PROXY_IMAGE is safely assumed, no need to have the default fallback, which could simplify code a bit.

**Review by h-w-chen — COMMENTED** [review]
not blocking others' approval.

File: cluster/addons/dns/coredns/coredns.yaml.base (hunk 1 above)
Comment (sonyafenge) [reply cluster/addons/dns/coredns/coredns.yaml.base:132]: This is a cherry-pick from Kubernetes: https://github.com/kubernetes/kubernetes/pull/82128.
Arktos Coredns did have same issue before cherry-pick.

File: cluster/gce/gci/scaleoutserver-helper.sh (hunk 3 above)
Comment (sonyafenge) [reply cluster/gce/gci/scaleoutserver-helper.sh]: thanks and updated.

File: cluster/gce/gci/configure.sh (hunk 2 above)
Comment (sonyafenge) [reply cluster/gce/gci/configure.sh:59]: Looks like new version of ubuntu build already add python installation. but anyway, we have conditions and if python already exist, will skip installation.

**Review by h-w-chen — APPROVED** [review]
lgtm

**Pull request merged**